turtlex_arm: # namespace. If this is commented, uncomment the namespace in the main launch file

    batch_size: 64  # maximum size of the batches sampled from memory
    replay_buffer_size: 20000

    training: True  # True for training, False for testing
    episodes_training: 400
    test_loops: 1  # number of times to loops through every goal position during testing
    max_episode_steps: 500

    load_model: False  # set the NUMBER or 'NAME' of the network to load, set False to skip loading (i.e. train from scratch)

    monitor: True  # Dedice whether to start the Gym Monitor wrapper or not

    # Turtlex-related parameters
    n_actions: 5 # future joint positions
    n_observations: 9 # Current joint positions (5) + goal position of the end effector (3) + distance of the end effector from the goal (1)

    joints_min_pos: [-2.617, -1.571, -1.571, -1.745, -2.617]
    joints_max_pos: [ 2.617,  1.571,  1.571,  1.745,  2.617]

    rounding_value: 2  # define how many decimal digits to allow in the observation vector

    init_joint_pos: # rest_arm position
      joint_1: 0.0
      joint_2: -1.0
      joint_3: 1.0
      joint_4: 1.2
      joint_5: 0.0

    ee_goals:
       x: [ 0.127, 0.205, 0.129,  0.29, 0.265, 0.288]
       y: [   0.0, 0.253,   0.0, 0.005,-0.125, 0.023]
       z: [ 0.328,  0.36, 0.299, 0.174, 0.336, 0.097]

    ee_bounds:
       x: [ -0.27, 0.36]
       y: [ -0.36, 0.36]
       z: [ 0.08, 0.63]

    max_distance: 1.0  # maximum distance of the EE to the desired GOAL EE

    world_name: 'office'

    score_hist_length: 30

    step_punishment: -1
    closer_reward: 10
    impossible_movement_punishment: -100
    reached_goal_reward: 100

# ee_goals: 
# 1 
#     position: 
#       x: 0.12606215427347997
#       y: 0.0003200067374127284
#       z: 0.32849627658762687
#     orientation: 
#       x: -0.0027209296241422825
#       y: 0.9909506137299954
#       z: 0.0006837298638354889
#       w: 0.13419765349513163
# 2
#     position: 
#       x: 0.2050265780795784
#       y: 0.00028735487632484735
#       z: 0.35967515346297135
#     orientation: 
#       x: 0.3428001419287651
#       y: 0.926027258213814
#       z: 0.055699768928416715
#       w: 0.14784828534665398
# 3
#     position: 
#       x: 0.12920304688382092
#       y: 0.25288309421287686
#       z: 0.2987819070947803
#     orientation: 
#       x: -0.5905672099775183
#       y: 0.7584357470077421
#       z: 0.12976026420200965
#       w: 0.24324444905157228
# 4
#     position: 
#       x: 0.290157111192081
#       y: 0.005148254015653095
#       z: 0.1741253879730031
#     orientation: 
#       x: -0.2775019272889635
#       y: 0.9593518888889913
#       z: -0.013066027637517831
#       w: 0.04965795563462985
# 5
#     position: 
#       x: 0.26515235246813423
#       y: -0.12506808963808919
#       z: 0.3357913825097993
#     orientation: 
#       x: -0.3847557683283364
#       y: 0.9084955223461344
#       z: -0.12538603174652452
#       w: 0.1042939483285364
# 6 - picking_arm
#     position: 
#       x: 0.2876205795129105
#       y: 0.022797669134486512
#       z: 0.09651705204317654
#     orientation: 
#       x: -0.04136336974828067
#       y: 0.998910728586725
#       z: 0.0010809603505478125
#       w: 0.02156987441145219
